bin/node/src/commands/discover.rs

```diff
@@ -142,7 +142,7 @@ impl Discovery {
                     }
                 }
                 _ = disc_ticker.tick() => {
-                    if let Ok(c) = handler.peer_count().try_recv() {
+                    if let Ok(c) = handler.peer_count().await {
```

Comment:
TODO: pull this into a spawned task that attempts to grab the pc and sends it over a channel here to await instead of using the `disc_ticker` and then blocking.